KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,314 @@
+
+    :param kls: The addendum that provides the words.
+    :return: Inputs results into a global list.
+    """
+
+    kls_list_chap = []
+    for item in kls:
+        if "paragraph{" in item or "subsubsection*{" in item:
+            kls_list.append(item[item.find("{") + 1: len(item) - 2])
+
+    kls_list = repeated_item_deleter(kls_list)
+
+    kls_list.append("Limit Relation")
+    for item in kls_list:
+        if item not in kls_list_chap:
+            kls_list_chap.append(item)
+    return kls_list_chap
+
+
+def fix_chapter_sort(kls, chap, word, sortloc, klsaddparas, sortmatch_2):
+    """
+    This function sorts through the input files and identifies and places sections from the addendum after their
+    respective subsections in the chapter.
+
+    :param kls: The addendum where the sections to be inserted are found.
+    :param chap: The destination of inserted sections.
+    :param word: The keyword that is being processed.
+    :param sortloc: The location in the list of words, of the word being sorted.
+    :return: This function outputs the processed chapter into a larger function for further processing.
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    sep1 = 0
+    name_chap = word.lower()
+    if name_chap == "orthogonality":
+        sep1 = 1
+    elif name_chap == "special value":
+        sep1 = 2
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    for item in kls:
+        index += 1
+        line = str(item)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12)+1: line.find("}", 12)]  # get just the name (like mathpeople)
+
+        if sep1 == 0:
+            if name_chap in line.lower() and kls.index(item) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line):
+                for item in klsaddparas:
+                    if index < item:
+                        klsloc = klsaddparas.index(item)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 1:
+            if name_chap in line.lower() and kls.index(item) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line) and "orthogonality relation" not in line.lower():
+                for item in klsaddparas:
+                    if index < item:
+                        klsloc = klsaddparas.index(item)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 2:
```

Comment:
A lot of this code (e.g. `elif sep1 == [int]: ...`) is very similar. Consider combining the code; there doesn't seem to be any reason for separation.